### PR TITLE
refactor + cleanup

### PR DESCRIPTION
**Pull request by DerCalli**

### CODE REVIEW

**Review by sourcecube — COMMENTED** [review]
IMHO `Discount` and `FilteredCalculator` could be done in composition, otherwise the classes get quite complex and are probably harder to update.

File: src/FondOfSpryker/Shared/Discount/Transfer/discount_custom_message.transfer.xml [hunk 1]
```diff
@@ -15,4 +15,8 @@
     <transfer name="DiscountConfigurator">
         <property name="discountCustomMessages" type="DiscountCustomMessage[]"/>
     </transfer>
+
+    <transfer name="Quote">
+        <property name="usedNotAppliedDiscount" type="Discount[]"/>
```
Comment (daniel-rose) [inline src/FondOfSpryker/Shared/Discount/Transfer/discount_custom_message.transfer.xml]: `<property name="usedNotAppliedDiscounts" singular="usedNotAppliedDiscount" type="Discount[]"/>` --> Would be better

File: src/FondOfSpryker/Zed/Discount/Business/DiscountFacade.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace FondOfSpryker\Zed\Discount\Business;
+
+use Spryker\Zed\Discount\Business\DiscountFacade as SprykerDiscountFacade;
+
+/**
+ * @method \FondOfSpryker\Zed\Discount\Business\DiscountBusinessFactory getFactory()
+ */
+class DiscountFacade extends SprykerDiscountFacade implements DiscountFacadeInterface
```
Comment (daniel-rose) [inline src/FondOfSpryker/Zed/Discount/Business/DiscountFacade.php]: Empty class!?

File: src/FondOfSpryker/Zed/Discount/Business/DiscountFacadeInterface.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace FondOfSpryker\Zed\Discount\Business;
+
+use Spryker\Zed\Discount\Business\DiscountFacadeInterface as SprykerDiscountFacadeInterface;
+
+interface DiscountFacadeInterface extends SprykerDiscountFacadeInterface
```
Comment (daniel-rose) [inline src/FondOfSpryker/Zed/Discount/Business/DiscountFacadeInterface.php]: Empty class!?

File: src/FondOfSpryker/Zed/Discount/Dependency/Form.php [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+
+class Form
```
Comment (daniel-rose) [inline src/FondOfSpryker/Zed/Discount/Dependency/Form.php]: Empty class!?

File: src/FondOfSpryker/Zed/Discount/Dependency/Persistence.php [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+
+class Persistence
```
Comment (daniel-rose) [inline src/FondOfSpryker/Zed/Discount/Dependency/Persistence.php]: Empty class!?

**Review by daniel-rose — CHANGES_REQUESTED** [review]

**Pull request merged**